Some of dask/dataframe/tests/test_rolling.py::test_time_rolling_methods are unreliable

**What happened**:

When running tests, some of the `test_time_rolling_methods` fail unreliably. Out of 10 test runs, 2 failed 2 tests, 4 failed 1 test and 4 succeeded.

Example failure:

```
=============================================================== test session starts ===============================================================
platform linux -- Python 3.6.11, pytest-6.0.1, py-1.9.0, pluggy-0.13.1
rootdir: /tmp/dask-2.22.0, configfile: setup.cfg
plugins: mock-3.2.0, pyfakefs-4.1.0, shutil-1.7.0, virtualenv-1.7.0, timeout-1.4.1, freezegun-0.4.2, localserver-0.5.0, forked-1.3.0, xdist-1.34.0, asyncio-0.14.0, hypothesis-5.23.7, flaky-3.7.0, httpbin-1.0.0
collected 48 items                                                                                                                                

dask/dataframe/tests/test_rolling.py ..................F...........F.................                                                       [100%]

==================================================================== FAILURES =====================================================================
__________________________________________________ test_time_rolling_methods[2S-std-args6-True] ___________________________________________________

method = 'std', args = (), window = '2S', check_less_precise = {}

    @pytest.mark.parametrize(
        "method,args,check_less_precise", rolling_method_args_check_less_precise
    )
    @pytest.mark.parametrize("window", ["1S", "2S", "3S", pd.offsets.Second(5)])
    def test_time_rolling_methods(method, args, window, check_less_precise):
        if dd._compat.PANDAS_GT_110:
            check_less_precise = {}
        else:
            check_less_precise = {"check_less_precise": check_less_precise}
    
        # DataFrame
        if method == "apply":
            kwargs = {"raw": False}
        else:
            kwargs = {}
        prolling = ts.rolling(window)
        drolling = dts.rolling(window)
        assert_eq(
            getattr(prolling, method)(*args, **kwargs),
            getattr(drolling, method)(*args, **kwargs),
>           **check_less_precise,
        )

dask/dataframe/tests/test_rolling.py:287: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
dask/dataframe/utils.py:817: in assert_eq
    tm.assert_frame_equal(a, b, **kwargs)
pandas/_libs/testing.pyx:64: in pandas._libs.testing.assert_almost_equal
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   AssertionError: DataFrame.iloc[:, 4] (column name="e") are different
E   
E   DataFrame.iloc[:, 4] (column name="e") values are different (2.5 %)
E   [index]: [2016-01-01T00:00:00.000000000, 2016-01-01T00:00:03.000000000, 2016-01-01T00:00:05.000000000, 2016-01-01T00:00:06.000000000, 2016-01-01T00:00:09.000000000, 2016-01-01T00:00:10.000000000, 2016-01-01T00:00:12.000000000, 2016-01-01T00:00:15.000000000, 2016-01-01T00:00:18.000000000, 2016-01-01T00:00:20.000000000, 2016-01-01T00:00:21.000000000, 2016-01-01T00:00:24.000000000, 2016-01-01T00:00:25.000000000, 2016-01-01T00:00:27.000000000, 2016-01-01T00:00:30.000000000, 2016-01-01T00:00:33.000000000, 2016-01-01T00:00:35.000000000, 2016-01-01T00:00:36.000000000, 2016-01-01T00:00:39.000000000, 2016-01-01T00:00:40.000000000, 2016-01-01T00:00:42.000000000, 2016-01-01T00:00:45.000000000, 2016-01-01T00:00:48.000000000, 2016-01-01T00:00:50.000000000, 2016-01-01T00:00:51.000000000, 2016-01-01T00:00:54.000000000, 2016-01-01T00:00:55.000000000, 2016-01-01T00:00:57.000000000, 2016-01-01T00:01:00.000000000, 2016-01-01T00:01:03.000000000, 2016-01-01T00:01:05.000000000, 2016-01-01T00:01:06.000000000, 2016-01-01T00:01:09.000000000, 2016-01-01T00:01:10.000000000, 2016-01-01T00:01:12.000000000, 2016-01-01T00:01:15.000000000, 2016-01-01T00:01:18.000000000, 2016-01-01T00:01:20.000000000, 2016-01-01T00:01:21.000000000, 2016-01-01T00:01:24.000000000]
E   [left]:  [nan, nan, nan, 26.870057685088806, nan, 34.64823227814084, nan, nan, nan, nan, 7.778174593052164, nan, 15.556349186104152, nan, nan, nan, nan, 34.64823227814085, nan, 10.606601717798219, nan, nan, nan, nan, 21.920310216782994, nan, 28.991378028648466, nan, nan, nan, nan, 24.748737341529182, nan, 9.899494936611735, nan, nan, nan, nan, 1.1680077279964342e-06, nan]
E   [right]: [nan, nan, nan, 26.870057685088806, nan, 34.64823227814084, nan, nan, nan, nan, 7.778174593052164, nan, 15.556349186104152, nan, nan, nan, nan, 34.64823227814083, nan, 10.606601717798192, nan, nan, nan, nan, 21.92031021678295, nan, 28.991378028648434, nan, nan, nan, nan, 24.748737341529164, nan, 9.899494936611688, nan, nan, nan, nan, 6.743495761743046e-07, nan]

pandas/_libs/testing.pyx:178: AssertionError
__________________________________________________ test_time_rolling_methods[3S-std-args6-True] ___________________________________________________

method = 'std', args = (), window = '3S', check_less_precise = {}

    @pytest.mark.parametrize(
        "method,args,check_less_precise", rolling_method_args_check_less_precise
    )
    @pytest.mark.parametrize("window", ["1S", "2S", "3S", pd.offsets.Second(5)])
    def test_time_rolling_methods(method, args, window, check_less_precise):
        if dd._compat.PANDAS_GT_110:
            check_less_precise = {}
        else:
            check_less_precise = {"check_less_precise": check_less_precise}
    
        # DataFrame
        if method == "apply":
            kwargs = {"raw": False}
        else:
            kwargs = {}
        prolling = ts.rolling(window)
        drolling = dts.rolling(window)
        assert_eq(
            getattr(prolling, method)(*args, **kwargs),
            getattr(drolling, method)(*args, **kwargs),
>           **check_less_precise,
        )

dask/dataframe/tests/test_rolling.py:287: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
dask/dataframe/utils.py:817: in assert_eq
    tm.assert_frame_equal(a, b, **kwargs)
pandas/_libs/testing.pyx:64: in pandas._libs.testing.assert_almost_equal
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   AssertionError: DataFrame.iloc[:, 4] (column name="e") are different
E   
E   DataFrame.iloc[:, 4] (column name="e") values are different (2.5 %)
E   [index]: [2016-01-01T00:00:00.000000000, 2016-01-01T00:00:03.000000000, 2016-01-01T00:00:05.000000000, 2016-01-01T00:00:06.000000000, 2016-01-01T00:00:09.000000000, 2016-01-01T00:00:10.000000000, 2016-01-01T00:00:12.000000000, 2016-01-01T00:00:15.000000000, 2016-01-01T00:00:18.000000000, 2016-01-01T00:00:20.000000000, 2016-01-01T00:00:21.000000000, 2016-01-01T00:00:24.000000000, 2016-01-01T00:00:25.000000000, 2016-01-01T00:00:27.000000000, 2016-01-01T00:00:30.000000000, 2016-01-01T00:00:33.000000000, 2016-01-01T00:00:35.000000000, 2016-01-01T00:00:36.000000000, 2016-01-01T00:00:39.000000000, 2016-01-01T00:00:40.000000000, 2016-01-01T00:00:42.000000000, 2016-01-01T00:00:45.000000000, 2016-01-01T00:00:48.000000000, 2016-01-01T00:00:50.000000000, 2016-01-01T00:00:51.000000000, 2016-01-01T00:00:54.000000000, 2016-01-01T00:00:55.000000000, 2016-01-01T00:00:57.000000000, 2016-01-01T00:01:00.000000000, 2016-01-01T00:01:03.000000000, 2016-01-01T00:01:05.000000000, 2016-01-01T00:01:06.000000000, 2016-01-01T00:01:09.000000000, 2016-01-01T00:01:10.000000000, 2016-01-01T00:01:12.000000000, 2016-01-01T00:01:15.000000000, 2016-01-01T00:01:18.000000000, 2016-01-01T00:01:20.000000000, 2016-01-01T00:01:21.000000000, 2016-01-01T00:01:24.000000000]
E   [left]:  [nan, nan, 16.97056274847714, 26.870057685088806, nan, 34.64823227814084, 42.42640687119286, nan, nan, 31.112698372208108, 7.778174593052169, nan, 15.556349186104155, 28.28427124746195, nan, nan, 2.121320343560413, 34.64823227814085, nan, 10.60660171779823, 43.13351365237942, nan, nan, 19.091883092036806, 21.92031021678299, nan, 28.991378028648466, 24.748737341529182, nan, nan, 4.242640687119392, 24.74873734152918, nan, 9.899494936611722, 4.949747468305956, nan, nan, 12.727922061357905, 1.1182822608526778e-06, nan]
E   [right]: [nan, nan, 16.97056274847714, 26.870057685088806, nan, 34.64823227814084, 42.42640687119286, nan, nan, 31.112698372208108, 7.778174593052169, nan, 15.556349186104155, 28.28427124746195, nan, nan, 2.1213203435596424, 34.64823227814083, nan, 10.606601717798192, 43.1335136523794, nan, nan, 19.091883092036756, 21.92031021678295, nan, 28.991378028648434, 24.748737341529147, nan, nan, 4.242640687119285, 24.74873734152916, nan, 9.899494936611676, 4.949747468305865, nan, nan, 12.72792206135787, 5.840038639982171e-07, nan]

pandas/_libs/testing.pyx:178: AssertionError
============================================================== slowest 10 durations ===============================================================
0.85s call     dask/dataframe/tests/test_rolling.py::test_time_rolling_methods[window3-apply-args11-False]
0.84s call     dask/dataframe/tests/test_rolling.py::test_time_rolling_methods[2S-apply-args11-False]
0.83s call     dask/dataframe/tests/test_rolling.py::test_time_rolling_methods[1S-apply-args11-False]
0.74s call     dask/dataframe/tests/test_rolling.py::test_time_rolling_methods[3S-apply-args11-False]
0.15s call     dask/dataframe/tests/test_rolling.py::test_time_rolling_methods[window3-std-args6-True]
0.14s call     dask/dataframe/tests/test_rolling.py::test_time_rolling_methods[window3-var-args7-True]
0.14s call     dask/dataframe/tests/test_rolling.py::test_time_rolling_methods[1S-count-args0-False]
0.14s call     dask/dataframe/tests/test_rolling.py::test_time_rolling_methods[2S-quantile-args10-False]
0.14s call     dask/dataframe/tests/test_rolling.py::test_time_rolling_methods[2S-var-args7-True]
0.14s call     dask/dataframe/tests/test_rolling.py::test_time_rolling_methods[1S-mean-args2-False]
========================================================== 2 failed, 46 passed in 9.89s ===========================================================
```

**What you expected to happen**:

I would like to see tests pass reliably on every run.

**Environment**:

- Dask version: 2.22.0
- Python version: 3.6.11, 3.7.8
- Operating System: Gentoo Linux
- Install method (conda, pip, source): source

--- a/dask/dataframe/tests/test_rolling.py
+++ b/dask/dataframe/tests/test_rolling.py
@@ -272,8 +272,8 @@ def test_time_rolling_methods(method, args, window, check_less_precise):
     if dd._compat.PANDAS_GT_110:
         if check_less_precise:
             check_less_precise = {"atol": 0.5e-3, "rtol": 0.5e-3}
-         else:
-             check_less_precise = {"atol": 0.5e-5, "rtol": 0.5e-5}
+        else:
+            check_less_precise = {"atol": 0.5e-5, "rtol": 0.5e-5}
     else:
         check_less_precise = {"check_less_precise": check_less_precise}
 

--- a/dask/dataframe/tests/test_rolling.py
+++ b/dask/dataframe/tests/test_rolling.py
@@ -270,7 +270,10 @@ def test_time_rolling_constructor():
 @pytest.mark.parametrize("window", ["1S", "2S", "3S", pd.offsets.Second(5)])
 def test_time_rolling_methods(method, args, window, check_less_precise):
     if dd._compat.PANDAS_GT_110:
-        check_less_precise = {"atol": 1e-4, "rtol": 1e-2}
+                if check_less_precise:
+                    check_less_precise = {"atol": 0.5e-3, "rtol": 0.5e-3}
+                 else:
+                     check_less_precise = {"atol": 0.5e-5, "rtol": 0.5e-5}
     else:
         check_less_precise = {"check_less_precise": check_less_precise}
 

--- a/dask/dataframe/tests/test_rolling.py
+++ b/dask/dataframe/tests/test_rolling.py
@@ -273,7 +273,7 @@ def test_time_rolling_methods(method, args, window, check_less_precise):
         if check_less_precise:
             check_less_precise = {"atol": 0.5e-3, "rtol": 0.5e-3}
         else:
-            check_less_precise = {"atol": 0.5e-5, "rtol": 0.5e-5}
+            check_less_precise = {}
     else:
         check_less_precise = {"check_less_precise": check_less_precise}
 

--- a/dask/dataframe/tests/test_rolling.py
+++ b/dask/dataframe/tests/test_rolling.py
@@ -270,10 +270,10 @@ def test_time_rolling_constructor():
 @pytest.mark.parametrize("window", ["1S", "2S", "3S", pd.offsets.Second(5)])
 def test_time_rolling_methods(method, args, window, check_less_precise):
     if dd._compat.PANDAS_GT_110:
-                if check_less_precise:
-                    check_less_precise = {"atol": 0.5e-3, "rtol": 0.5e-3}
-                 else:
-                     check_less_precise = {"atol": 0.5e-5, "rtol": 0.5e-5}
+        if check_less_precise:
+            check_less_precise = {"atol": 0.5e-3, "rtol": 0.5e-3}
+         else:
+             check_less_precise = {"atol": 0.5e-5, "rtol": 0.5e-5}
     else:
         check_less_precise = {"check_less_precise": check_less_precise}
 

--- a/dask/dataframe/tests/test_rolling.py
+++ b/dask/dataframe/tests/test_rolling.py
@@ -270,7 +270,7 @@ def test_time_rolling_constructor():
 @pytest.mark.parametrize("window", ["1S", "2S", "3S", pd.offsets.Second(5)])
 def test_time_rolling_methods(method, args, window, check_less_precise):
     if dd._compat.PANDAS_GT_110:
-        check_less_precise = {}
+        check_less_precise = {"atol": 1e-4, "rtol": 1e-2}
     else:
         check_less_precise = {"check_less_precise": check_less_precise}
 

